traffic_ops/traffic_ops_golang/trafficvault/backends/postgres/postgres.go

```diff
@@ -199,12 +215,10 @@ func (p *Postgres) GetExpirationInformation(tx *sql.Tx, ctx context.Context, day
 		if err = rows.Scan(&expirationInfo.DeliveryService, &expirationInfo.CDN, &expirationInfo.Provider, &expirationInfo.Expiration); err != nil {
 			return []tc.SSLKeyExpirationInformation{}, err
 		}
-		expirationInfo.Federated = false
-		for _, fed := range fedList {
-			if fed == expirationInfo.DeliveryService {
-				expirationInfo.Federated = true
-			}
+		if inactiveList[expirationInfo.DeliveryService] {
+			continue
 		}
+		expirationInfo.Federated = util.ContainsStr(fedList, expirationInfo.DeliveryService)
```

Comment:
I should have explained better in my previous comment, my bad. But, do you think converting the `fedList` into a map also makes sense, because of the same reason? That way we won't have to traverse the `fedList` array to look for the `expirationInfo.DeliveryService` key.